src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/LocalDebuggerCommandArgumentsValueProvider.cs

```diff
@@ -0,0 +1,74 @@
+    internal sealed class LocalDebuggerCommandArgumentsValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private readonly Lazy<ProjectProperties> _projectProperties;
+        private readonly Lazy<IOutputGroupsService, IAppliesToMetadataView> _outputGroups;
+        private readonly ConfiguredProject _configuredProject;
+
+        [ImportingConstructor]
+        public LocalDebuggerCommandArgumentsValueProvider(Lazy<ProjectProperties> projectProperties, 
+                                                          [Import(AllowDefault = true)] Lazy<IOutputGroupsService, IAppliesToMetadataView> outputGroups,
+                                                          ConfiguredProject configuredProject)
+        {
+            Requires.NotNull(projectProperties, nameof(projectProperties));
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+
+            _projectProperties = projectProperties;
+            _outputGroups = outputGroups;
+            _configuredProject = configuredProject;
+        }
+
+        private Lazy<IOutputGroupsService> OutputGroups {
+            get {
+                return _outputGroups != null && _outputGroups.AppliesTo(_configuredProject.Capabilities) ? _outputGroups : null;
+            }
+        }
+
+        public async override Task<string> OnGetEvaluatedPropertyValueAsync(string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            // We do the same check as in the debugger command, so empty string for the local debugger properties means we should perform
+            // the interception.
+            var debugCommand = await defaultProperties.GetEvaluatedPropertyValueAsync(WindowsLocalDebugger.LocalDebuggerCommandProperty).ConfigureAwait(false);
+            var commandArgs = evaluatedPropertyValue;
+
+            if (debugCommand == LocalDebuggerCommandValueProvider.DefaultCommand)
+            {
+                // Get the path of the executable, and plug it into "exec path original_args"
+                var executable = await GetExecutablePath().ConfigureAwait(false);
+                commandArgs = $"exec {executable} {commandArgs}";
+            }
+
+            return commandArgs;
+        }
+
+        private async Task<string> GetExecutablePath()
+        {
+            var generalProperties = await _projectProperties.Value.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
+            var command = await generalProperties.TargetPath.GetEvaluatedValueAsync().ConfigureAwait(false);
+
+            if (string.IsNullOrEmpty(command) && OutputGroups != null)
+            {
+                command = await OutputGroups.Value.GetKeyOutputAsync().ConfigureAwait(false);
+            }
+
+            // Because a .NET Core app produces an executable dll, and not an actual executable, we need to change the extension to be .dll
+            var rawName = Path.GetFileNameWithoutExtension(command);
```

Comment:
Secondly, these properties come from the project file - they can we anything, so in the future we'll want to guard against bad paths, etc.